### ISSUE TITLE
loss interpretation

### ISSUE THREAD

**Issue by hxngiee** [samxuxiang/BrepGen#5]
Hi @samxuxiang,

When training a surface latent model, I wonder if it is true that the [loss curve](https://github.com/samxuxiang/BrepGen/blob/1b54431725d6ec44099af902384add1d6dbb411e/trainer.py#L595C1-L596C1) in test_val increases as training progresses.
Shouldn't we use checkpoint file in a step with less loss?
<img width="1990" alt="image" src="https://github.com/samxuxiang/BrepGen/assets/76642977/0661afdc-cf3b-49af-95d6-ed3bef51ee95">

Also, is there a reason to check the timestep starting from the lowest value?

**Comment by samxuxiang** [comment]
Hi @hxngiee , due to the relative small size of the furniture data, on some categories train/test objects will be quite different and  validation loss can increase due to overfitting. Categories like chair/table will more data will be less affected. Also the validation loss for deepcad/abc will continue to decrease as the dataset is much larger. 
![Selection_001](https://github.com/samxuxiang/BrepGen/assets/22716231/370c4396-6c15-4b5f-9757-67756df53a08)

**Comment by hxngiee** [comment]
Thank you so much for your clarification @samxuxiang 

This project seems to be very well-structured and impressive.

If possible, I would like to see a visualization of the denoising process. 

Do you have any references or examples that I could check out?

**Comment by samxuxiang** [comment]
Hi @hxngiee , you can store the intermediate decoded points at different step of the denoising, and then render them using [BlenderToolbox](https://github.com/HTDerekLiu/BlenderToolbox).
